### DISCUSSION TITLE
Is there a sharpening implementation similar to USM

### DISCUSSION THREAD

**Discussion by shijiege** [libvips/libvips#3916] [General]
i try to use VIPS_sharpen() ，but its effect can not close to imageMagic unsharp.so i want know Is there a sharpening implementation similar to USM, or the best vips_sharpen() settings for one pic

**Comment by jcupitt** [comment]
Hello @shijiege,

`vips_sharpen()` is a cored luminance usm, so it's more sophisticated than the usm in imagemagick. You ought to see improved quality.

[nip2](https://github.com/libvips/nip2) has some GUI for it, it might make it easier to tune the parameters. Load an image, select Filter / Convolution / Unsharp, then adjust the sliders. You can open the output image and zoom in on details while you adjust the controls.

![image](https://github.com/libvips/libvips/assets/580843/40087b9a-b538-47ee-ae5c-45f70a042259)
